ParameterLib/Utils.h

```diff
@@ -68,6 +70,19 @@ Parameter<ParameterDataType>* findParameterOptional(
             num_components);
     }
 
+    // Test the parameter's mesh only if there is a "test"-mesh provided.
+    if (mesh != nullptr)
+    {
+        if (auto const error = isDefinedOnSameMesh(*parameter, *mesh))
+        {
+            OGS_FATAL(
+                "The found parameter is not suitable for the use on the "
+                "required mesh.\n%s",
+                *error);
```

Comment:
Is it correct that `*error` is a string? I thought it is necessary to use the `c_str()` method for output?

Reply:
That's right; in the message there is a `%s`, and `*error` is a string argument.